### PR TITLE
ESP backport manual receive BFW-2166

### PR DESCRIPTION
**Pull request by vladamatena**
- Port ESP 1.x FW support to recent enough LwESP supporting TCP manual receive
- Various LwESP integration fixes
- Fixes two ALTCP - LwESP integration layer bugs

### CODE REVIEW

File: lib/ESP/lwesp/.gitrepo [hunk 1]
```diff
@@ -4,9 +4,9 @@
 ; git-subrepo command. See https://github.com/git-commands/git-subrepo#readme
 ;
 [subrepo]
-	remote = https://github.com/MaJerle/lwesp.git
-	branch = v0.5.0
-	commit = 4dd5399fabebedf601042d5dad4c29f9e6c24abe
-	parent = 04f8547400d0f5493dbd8f690b24146b5d144999
+	remote = https://github.com/vladamatena/lwesp.git
```
Comment (vorner) [inline lib/ESP/lwesp/.gitrepo:7]: Question about this subrepo. I assume it contains some of the backports. Are we going to maintain them, or will they be eventually pushed upstream? If the former, should it live under your account, or should it move to `prusa3d`?

File: lib/WUI/netdev.c [hunk 2]
```diff
@@ -184,7 +200,7 @@ uint32_t netdev_init() {
 }
 
 uint32_t netdev_init_esp() {
-    esp_hard_reset_device();
+    // esp_hard_reset_device();
```
Comment (vorner) [inline lib/WUI/netdev.c:203]: Any reason why we are not resetting the device here, to make sure it starts in known state?

File: lib/ESP/prusa/src/system/lwesp_ll_buddy.c [hunk 3]
```diff
@@ -87,10 +87,11 @@ void StartUartBufferThread(void const *arg) {
     }
 }
 
-void esp_hard_reset_device() {
+uint8_t esp_hard_reset_device(uint8_t state) {
```
Comment (vorner) [inline lib/ESP/prusa/src/system/lwesp_ll_buddy.c:90]: Is `state` used for anything?

**Review by vorner — COMMENTED** [review]
Is there a way to see what changes you did and which were introduced by pulling from some upstream?

File: lib/WUI/netdev.c (hunk 2 above)
Comment (vladamatena) [reply lib/WUI/netdev.c:203]: Leaving commented out code behind was a mistake. Integrated upstream changes in LwESP implement hard reset as part of init sequence. We just provide a function pointer to do it.

**Pull request merged**